java/vector/src/test/java/org/apache/arrow/vector/ipc/TestJSONFile.java

```diff
@@ -429,6 +429,27 @@ public void testWriteReadNullJSON() throws IOException {
     }
   }
 
+  @Test
+  public void testWriteReadExtensionData() throws IOException {
+    // write
+    try (VectorSchemaRoot root = writeUuidData(allocator)) {
+      File file = new File("target/mytest_extension.json");
+      writeJSON(file, root, null);
+    }
+
+    // read
+    try (BufferAllocator readerAllocator = allocator.newChildAllocator("reader", 0, Integer.MAX_VALUE);
+        JsonFileReader reader = new JsonFileReader(new File("target/mytest_extension.json"), readerAllocator)) {
+      Schema schema = reader.start();
+      LOGGER.debug("reading schema: " + schema);
```

Comment:
nit: necessary?

Reply:
Quite possibly not, but I ~copy-pasted~ took inspiration from the other tests in the file - happy to remove?